rules/Naming/Rector/Class_/RenamePropertyToMatchTypeRector.php

```diff
@@ -148,6 +148,11 @@ private function skipDateTimeOrMockObjectPropertyType(Property $property): bool
         if ($this->isName($property->type, ClassName::MOCK_OBJECT)) {
             return true;
         }
+
+        if ($this->isName($property->type, ClassName::DATE_TIME_INTERFACE)) {
```

Comment:
I think using `isObjectType()` is better than manual definition of it, eg;

```php
if ($this->isObjectType($property->type, new \PHPStan\Type\ObjectType(ClassName::DATE_TIME_INTERFACE)) {
    return true;
}
```

then `DateTime`, `DateTimeImmutable` will be skipped as well, see example

https://github.com/rectorphp/rector-src/blob/f5edbf51d46fe03b987f2573e900d53c8370d5bc/rules/DeadCode/Rector/ClassLike/RemoveTypedPropertyNonMockDocblockRector.php#L101